src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -170,9 +168,16 @@
         logger.info("Order successfully retrieved from storage");
         return validatePoNumber(poFromStorage, compPO)
           .thenCompose(v -> updatePoLines(poFromStorage, compPO))
```

Comment:
`updatePoLines` updates order lines in the storage but `checkOrderApprovalPermissions` might throw exception. Probably the logic to check approval permissions should be invoked first

Reply:
Done